src/main/webapp/script.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Event processing for upload button
+ */
+function fileUploadDisplay() {
+  document.querySelector('.bg-model').style.display = 'flex';
+}
+
+/**
+ * Event processing for close button
+ */
+function closeDiv() {
+  document.querySelector('.bg-model').style.display = 'none';
+}
+
+/**
+ * Makes the form for uploading an image visible and adds the 'action' to it by fetching
+ * the url(that the form needs to post to) from the servlet
+ */
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url')
+      .then((response) => response.text())
+      .then((imageUploadUrl) => {
+        const uploadForm = document.getElementById('upload-barcode-form');
+        uploadForm.action = imageUploadUrl;
+        uploadForm.classList.remove('hidden');
+      });
+}
+
+/**
+ * Fetches all image keys stored in the database and
+ * displays them on the web page along with date, time and message
+ * using get-image-url servlet
+ */
+function getAndShowImages() {
+  const imageListElement = document.getElementById('history-images');
+
+  fetch('/handle-image', {
+    headers: {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json',
+    },
+  })
+      .then((response) => response.json()).then((imageDetails) => {
+        if (imageDetails.length == 0) {
+          imageListElement.innerText = 'No history';
+        } else {
+          imageDetails.forEach((imageDetail) => {
+            imageListElement.appendChild(createImageElement(imageDetail));
+          });
```

Comment:
For testing this is fine, but later on you probably don't want to show all user's the history of all other users.

Reply:
True that, we were planning to use authentication system and store each user's history and stuff separately. Also like we did with the comments in our portfolios, we can let user select the option of how many history searches should be shown. 
OR
We can simply place all the history in an .overflow div.

Reply:
SGTM